### ISSUE TITLE
No instance for (Data.Default.Default Request)

### ISSUE THREAD

**Issue by juhp** [sethfowler/hsbugzilla#1]
I having trouble building bugzilla-0.2.1.0 (with ghc-7.6.3):

```
[2 of 5] Compiling Web.Bugzilla.Internal.Network ( src/Web/Bugzilla/Internal/Network.hs, dist/build/Web/Bugzilla/Internal/Network.o )

src/Web/Bugzilla/Internal/Network.hs:73:3:
    No instance for (Data.Default.Default Request)
      arising from a use of `def'
    Possible fix:
      add an instance declaration for (Data.Default.Default Request)
    In the expression: def
    In the expression: def {secure = True, port = 443}
    In an equation for `sslRequest':
        sslRequest = def {secure = True, port = 443}
Failed to install bugzilla-0.2.1.0
```

**Comment by sethfowler** [comment]
Thanks for the bug report!

The simplest explanation I can think of is that you're somehow building against an older version of `http-conduit`, but the package's version bounds should prevent that, so I'm not sure how it could happen.

Would you mind posting the full output of `cabal install -v bugzilla`?

**Comment by juhp** [comment]
Well I just tried building it in a sandbox and seemed to run into the same error.
The following build deps got pulled into the sandbox:

```
aeson-0.7.0.6          crypto-numbers-0.2.3         pem-0.2.2
asn1-encoding-0.8.1.3  crypto-pubkey-0.2.4          publicsuffixlist-0.1
asn1-parse-0.8.1       crypto-pubkey-types-0.4.2.2  scientific-0.3.2.1
asn1-types-0.2.3       crypto-random-0.0.7          socks-0.5.4
attoparsec-0.11.3.4    cryptohash-0.11.6            tls-1.2.8
cereal-0.4.0.1         http-client-0.3.3.2          x509-1.4.11
cipher-aes-0.2.7       http-client-conduit-0.2.0.1  x509-store-1.4.4
cipher-des-0.0.6       http-client-tls-0.2.1.2      x509-system-1.4.5
cipher-rc4-0.1.4       http-conduit-2.0.0.10        x509-validation-1.5.0
connection-0.2.1       iso8601-time-0.1.2
cprng-aes-0.5.2        mime-types-0.1.0.4
```

Where should `def` be defined actually?  I couldn't find any Default Request instance.

Anyway here is the verbose build output:

```
$ cabal-dev install -v
/usr/bin/cabal --numeric-version
/usr/bin/ghc-pkg --version
Using ghc-pkg 7.6.3
/usr/bin/cabal --version
/usr/bin/cabal --config-file=/home/petersen/haskell/bugzilla-0.2.1.0/cabal-dev/cabal.config install --verbose=2
Reading available packages...
Choosing modular solver.
Resolving dependencies...
Ready to install bugzilla-0.2.1.0
Waiting for install task to finish...
Configuring bugzilla-0.2.1.0...
Flags chosen: builddemo=False
Dependency aeson ==0.7.0.6: using aeson-0.7.0.6
Dependency base ==4.6.0.1: using base-4.6.0.1
Dependency blaze-builder ==0.3.1.1: using blaze-builder-0.3.1.1
Dependency bugzilla -any: using bugzilla-0.2.1.0
Dependency bytestring ==0.10.0.2: using bytestring-0.10.0.2
Dependency connection ==0.2.1: using connection-0.2.1
Dependency containers ==0.5.0.0: using containers-0.5.0.0
Dependency data-default ==0.5.1: using data-default-0.5.1
Dependency http-conduit ==2.0.0.10: using http-conduit-2.0.0.10
Dependency http-types ==0.8.0: using http-types-0.8.0
Dependency iso8601-time ==0.1.2: using iso8601-time-0.1.2
Dependency resourcet ==0.4.10.2: using resourcet-0.4.10.2
Dependency text ==0.11.3.1: using text-0.11.3.1
Dependency time ==1.4.0.1: using time-1.4.0.1
Dependency transformers ==0.3.0.0: using transformers-0.3.0.0
Dependency unordered-containers ==0.2.3.0: using unordered-containers-0.2.3.0
Dependency vector ==0.10.0.1: using vector-0.10.0.1
Using Cabal-1.16.0 compiled by ghc-7.6
Using compiler: ghc-7.6.3
Using install prefix: /home/petersen/haskell/bugzilla-0.2.1.0/cabal-dev/
Binaries installed in: /home/petersen/haskell/bugzilla-0.2.1.0/cabal-dev//bin
Libraries installed in:
/home/petersen/haskell/bugzilla-0.2.1.0/cabal-dev//lib/bugzilla-0.2.1.0/ghc-7.6.3
Private binaries installed in:
/home/petersen/haskell/bugzilla-0.2.1.0/cabal-dev//libexec
Data files installed in:
/home/petersen/haskell/bugzilla-0.2.1.0/cabal-dev//share/bugzilla-0.2.1.0
Documentation installed in:
/home/petersen/haskell/bugzilla-0.2.1.0/cabal-dev//share/doc/bugzilla-0.2.1.0
Using alex version 3.1.0 found on system at: /home/petersen/.cabal/bin/alex
Using ar found on system at: /usr/bin/ar
No c2hs found
Using cpphs version 1.18.4 found on system at: /home/petersen/.cabal/bin/cpphs
No ffihugs found
Using gcc version 4.8.2 found on system at: /usr/bin/gcc
Using ghc version 7.6.3 found on system at: /usr/bin/ghc
Using ghc-pkg version 7.6.3 found on system at: /usr/bin/ghc-pkg
No greencard found
Using haddock version 2.13.2 found on system at: /usr/bin/haddock
Using happy version 1.19.0 found on system at: /home/petersen/.cabal/bin/happy
No hmake found
Using hpc version 0.6 found on system at: /usr/bin/hpc
Using hsc2hs version 0.67 found on system at: /usr/bin/hsc2hs
Using hscolour version 1.20 found on system at: /usr/bin/HsColour
No hugs found
No jhc found
Using ld found on system at: /usr/bin/ld
No lhc found
No lhc-pkg found
No nhc98 found
Using pkg-config version 0.28 found on system at: /usr/bin/pkg-config
Using ranlib found on system at: /usr/bin/ranlib
Using strip found on system at: /usr/bin/strip
Using tar found on system at: /usr/bin/tar
No uhc found
creating dist/build
creating dist/build/autogen
Building bugzilla-0.2.1.0...
Preprocessing library bugzilla-0.2.1.0...
Building library...
creating dist/build
/usr/bin/ghc --make -fbuilding-cabal-package -O -odir dist/build -hidir dist/build -stubdir dist/build -i -idist/build -isrc -idist/build/autogen -Idist/build/autogen -Idist/build -optP-include -optPdist/build/autogen/cabal_macros.h -package-name bugzilla-0.2.1.0 -hide-all-packages -no-user-package-db -package-db /home/petersen/haskell/bugzilla-0.2.1.0/cabal-dev/packages-7.6.3.conf -package-db dist/package.conf.inplace -package-id aeson-0.7.0.6-d0f9daa706ba1f16996b2ceec09842b6 -package-id base-4.6.0.1-8aa5d403c45ea59dcd2c39f123e27d57 -package-id blaze-builder-0.3.1.1-71a2ae53eb5daf9e6673db8b47b3b6b5 -package-id bytestring-0.10.0.2-4f93248f75667c2c3321a7a6761b576f -package-id connection-0.2.1-7afb06e9fbb3e2b598bf74da1a7f1f8e -package-id containers-0.5.0.0-ab1dae9a94cd3cc84e7b2805636ebfa2 -package-id data-default-0.5.1-3dcf8b1e18d6ff182bcfe57129991c1d -package-id http-conduit-2.0.0.10-d19f170c807294044fd7b1115fe52410 -package-id http-types-0.8.0-6b360a20eef84a16b35dc479214b5b74 -package-id iso8601-time-0.1.2-7fad6fcf7eb7dc1a2cd677674bd1fb46 -package-id resourcet-0.4.10.2-8693cff92f1f1f2799f068c36d9c1700 -package-id text-0.11.3.1-e38859e86485c167fa7c9441789e7607 -package-id time-1.4.0.1-10dc4804a19dc0000fab79908f1a9f50 -package-id transformers-0.3.0.0-387c76a892254b0b8fe4d66f4780ad17 -package-id unordered-containers-0.2.3.0-c7952fe8bed178bdf8ed4d090730c9da -package-id vector-0.10.0.1-869166d5d49db46ce6c328ea5f8defbf -XHaskell2010 Web.Bugzilla Web.Bugzilla.Search Web.Bugzilla.Internal.Network Web.Bugzilla.Internal.Search Web.Bugzilla.Internal.Types -Wall
[2 of 5] Compiling Web.Bugzilla.Internal.Network ( src/Web/Bugzilla/Internal/Network.hs, dist/build/Web/Bugzilla/Internal/Network.o )

src/Web/Bugzilla/Internal/Network.hs:73:3:
    No instance for (Data.Default.Default Request)
      arising from a use of `def'
    Possible fix:
      add an instance declaration for (Data.Default.Default Request)
    In the expression: def
    In the expression: def {secure = True, port = 443}
    In an equation for `sslRequest':
        sslRequest = def {secure = True, port = 443}
Failed to install bugzilla-0.2.1.0
cabal: Error: some packages failed to install:
bugzilla-0.2.1.0 failed during the building phase. The exception was:
ExitFailure 1
```

**Comment by sethfowler** [comment]
Thanks for the verbose build output.

I'm not sure what's going wrong. The version of `http-conduit` you're using certainly provides a `Default` instance for `Request`, which the docs [tell you to use](http://hackage.haskell.org/package/http-conduit-2.0.0.10/docs/Network-HTTP-Conduit.html#g:3).

I use this package on a machine which runs GHC 7.6.3. I checked and found that it's using `http-conduit 2.0.0.10`, which is the same version you're installing, but on my system the `bugzilla` package was built with`data-default 0.5.3`. My best guess at this point is that the version bounds I've used aren't tight enough on `data-default`.

If you modify `bugzilla.cabal` to require `data-default ==0.5.3`, does that fix the problem?
